_source/_posts/2023-07-27-spring-graphql-react.adoc

```diff
@@ -0,0 +1,1857 @@
+
+        this.graphQlTester
+                .documentName("companyList")
+                .variable("page", 0)
+                .execute()
+                .path("companyList")
+                .matchesJson("""
+                    [{
+                        "id": null,
+                        "SIC": "1234",
+                        "name": "Test Company",
+                        "status": "active",
+                        "category": "private",
+                        "companyNumber": "12345678",
+                        "countryOfOrigin": "UK"
+                    }]
+                """);
+    }
+}
+----
+
+Create the document file `companyList.graphql` containing the query definition for the test, in the folder `src/main/test/resources/graphql-test`:
+
+.companyList.graphql
+[source, graphql]
+----
+query companyList($page: Int) {
+    companyList(page: $page) {
+        id
+        SIC
+        name
+        status
+        category
+        companyNumber
+        countryOfOrigin
+    }
+}
+----
+Update the test configuration in `build.gradle` file, so passed tests are logged:
+
+.build.gradle
+[source, groovy]
+----
+tasks.named('test') {
+  useJUnitPlatform()
+
+  testLogging {
+    // set options for log level LIFECYCLE
+    events "failed", "passed"
+  }
+}
+----
+
+Run the test with:
+
+[source, shell]
+----
+./gradlew clean test
+----
+You should see logs for the successful tests:
+
+
+----
+...
+SpringBootHraphQlApiApplicationTests > contextLoads() PASSED
+
+CompanyControllerTests > shouldGetCompany() PASSED
+...
+----
+
+
+=== Add Neo4j seed data
+
+Let's add Neo4j migrations dependency for the seed data insertion. Edit the file `build.gradle` and add:
+
+.build.gradle
+[source, groovy]
+----
+dependencies {
+  ...
+  implementation 'eu.michael-simons.neo4j:neo4j-migrations-spring-boot-starter:2.4.0'
+}
+----
+
+Create the folder `src/main/resources/neo4j/migrations` and the following migration files:
+
+.V001__Constraint.cypher
+[source, cypher]
+----
+{% raw %}
+CREATE CONSTRAINT FOR (c:Company) REQUIRE c.companyNumber IS UNIQUE;
+//Constraint for a node key is a Neo4j Enterprise feature only - run on an instance with enterprise
+//CREATE CONSTRAINT ON (p:Person) ASSERT (p.birthMonth, p.birthYear, p.name) IS NODE KEY
+CREATE CONSTRAINT FOR (p:Property) REQUIRE p.titleNumber IS UNIQUE;
+{% endraw %}
+----
+
+.V002__Company.cypher
+[source, cypher]
+----
+{% raw %}
+LOAD CSV WITH HEADERS FROM "file:///PSCAmericans.csv" AS row
+MERGE (c:Company {companyNumber: row.company_number})
+RETURN COUNT(*);
+{% endraw %}
+----
+
+.V003__Person.cypher
+[source, cypher]
+----
+{% raw %}
+LOAD CSV WITH HEADERS FROM "file:///PSCAmericans.csv" AS row
+MERGE (p:Person {name: row.`data.name`, birthYear: row.`data.date_of_birth.year`, birthMonth: row.`data.date_of_birth.month`})
+  ON CREATE SET p.nationality = row.`data.nationality`,
+  p.countryOfResidence = row.`data.country_of_residence`
+// TODO: Address
```

Comment:
Is this relevant? If so, does it get taken care of below?

Reply:
Removed, not addressed